src/main/java/co/rsk/bitcoinj/script/RedeemScriptParserFactory.java

```diff
@@ -86,17 +82,11 @@ private static ParseResult extractRedeemScriptFromChunks(List<ScriptChunk> chunk
                     lastByte == ScriptOpCodes.OP_CHECKMULTISIGVERIFY ||
                     lastByte == ScriptOpCodes.OP_ENDIF
             ) {
-                ScriptParserResult result = ScriptParser.parseScriptProgram(lastChunk.data);
-                if (result.getException().isPresent()) {
-                    String message = String.format("Error trying to parse inner script. %s", result.getException().get());
-                    logger.debug("[extractRedeemScriptFromChunks] {}", message);
-                    throw new ScriptException(message);
-                }
-                return new ParseResult(result.getChunks());
+                return ScriptParser.parseScriptProgram(lastChunk.data);
             }
         }
         logger.debug("[extractRedeemScriptFromChunks] Could not get redeem script from given chunks");
-        return null;
+        throw new ScriptException("Could not get redeem script from given chunks");
```

Comment:
You could put the message in a variable and reuse it

Reply:
Updated